core/services/eth/log_broadcaster.go

```diff
@@ -141,12 +141,16 @@ func (b *logBroadcaster) Unregister(address common.Address, listener LogListener
 // notifies its subscribers.
 func (b *logBroadcaster) startResubscribeLoop() {
 	defer close(b.chDone)
+	var subscription eth.Subscription = noopSubscription{}
+	chRawLogs := make(chan eth.Log)
+	defer func() { subscription.Unsubscribe() }()
+	defer func() { close(chRawLogs) }()
```

Comment:
Does `subscription.Unsubscribe()` definitely not close the channel?  Guess that makes sense, since it has to be created by the calling code.

If this is the case, then I'm wondering if it wouldn't make more sense to put the channel creation inside of `LogBroadcaster#SubscribeToLogs` and put the channel closure inside of `Unsubscribe`.  This would require tweaking some of the tests, but probably not much work.  What do you think?